src/Containers/Microsoft.NET.Build.Containers/Registry/DefaultBlobOperations.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+
+namespace Microsoft.NET.Build.Containers.Registry;
+
+internal class DefaultBlobOperations : IBlobOperations
+{
+    private HttpClient Client { get; }
+    public DefaultBlobOperations(HttpClient client)
+    {
+        Client = client;
+        Upload = new DefaultBlobUploadOperations(Client);
+    }
+
+    public IBlobUploadOperations Upload { get; }
+
+    public async Task<HttpResponseMessage> GetAsync(string repositoryName, string digest, CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        using var request = new HttpRequestMessage(HttpMethod.Get, new Uri($"/v2/{repositoryName}/blobs/{digest}"));
```

Comment:
@baronfel there is a problem with this code now, previously this line was
`using var request = new HttpRequestMessage(HttpMethod.Get, new Uri(BaseUri, $"/v2/{repositoryName}/manifests/{reference}"));`

Now, `BaseUri` is missing. There are number of tests failing because of that.  Could you please take a look?

Otherwise, I will start on this tomorrow.